### PR TITLE
feat: Add docs on how to use private Docker workspace images

### PR DESCRIPTION
**Pull request by atduarte**
## Description

Document https://github.com/gitpod-io/gitpod/pull/8550

## Related Issue(s)

Fixes https://github.com/gitpod-io/website/issues/1704

## How to test

Check preview.

## Release Notes

```release-note
Documented private Docker registries support
```

<a href="https://gitpod-staging.com/#https://github.com/gitpod-io/website/pull/2951"></a>

<a href="https://gitpod.io/#https://github.com/gitpod-io/website/pull/2951"></a>

### CODE REVIEW

File: gitpod/docs/configure/workspaces/workspace-image.md [hunk 1]
```diff
@@ -33,7 +33,17 @@ For public images, feel free to specify a tag, e.g. `image: node:buster` if you
...
 For Gitpod images, we recommend using timestamped tag for maximum reproducibility, for example `image: gitpod/workspace-full:2022-05-08-14-31-53` (taken from the `Tags` panel on [this dockerhub page](https://hub.docker.com/r/gitpod/workspace-full/tags) for example)
 
-## Configure a custom Dockerfile
+## Use a private Docker image
+
+You may also use private Docker images.
+
+To do so you must provide the registry authentication details to Gitpod by setting the `GITPOD_IMAGE_AUTH` environment variable in the following format: `<registry-domain>:<base64-encoded 'username:password'>`.
+
+For example, if the registry is `docker.io`, the username is `foo` and the password is `bar`, the `GITPOD_IMAGE_AUTH` environment variable value may be calculated using the command `echo -n "docker.io:"; echo -n "foo:bar" | base64 -w0` which outputs `docker.io:Zm9vOmJhcg==`.
```
Comment (kylos101) [inline gitpod/docs/configure/workspaces/workspace-image.md:42]: Out of curiosity, which private registry did you test with? I assume you tested by setting the env var as a Project level variable?

**Review by kylos101 — COMMENTED** [review]
Thank you so much, @atduarte ! Just a couple questions to firm up how you tested.

**Comment by Siddhant-K-code** [comment]
We can use [this repo](https://gitlab.com/mrzarquon/private-image) (it contains some docs & steps) to test it out.

File: gitpod/docs/configure/workspaces/workspace-image.md (hunk 1 above)
Comment (atduarte) [reply gitpod/docs/configure/workspaces/workspace-image.md:42]: Indeed I didn't test User-level variables. It doesn't seem to work.

Why are they different?

File: gitpod/docs/configure/workspaces/workspace-image.md [hunk 2]
```diff
@@ -33,7 +33,17 @@ For public images, feel free to specify a tag, e.g. `image: node:buster` if you
...
+
+> **Reminder:** Environment variables can be set at the user or project level. You can read more [here](/docs/configure/projects/environment-variables)
```
[the remark this answers is not in the document]
Comment (atduarte) [reply gitpod/docs/configure/workspaces/workspace-image.md]: Indeed I didn't test User-level variables. It doesn't seem to work.

Why are they different?

File: gitpod/docs/configure/workspaces/workspace-image.md (hunk 1 above)
Comment (atduarte) [reply gitpod/docs/configure/workspaces/workspace-image.md:44]: Gitlab's

File: gitpod/docs/configure/workspaces/workspace-image.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (kylos101) [reply gitpod/docs/configure/workspaces/workspace-image.md]: :shrug: Good question! Perhaps an opportunity for the future?

**Review by kylos101 — APPROVED** [review]
:shipit: thanks, @atduarte !!!

:wave: @Siddhant-K-code

**Comment by Siddhant-K-code** [comment]
Thanks @atduarte @kylos101 for this 🧡

**Pull request merged**